chasm/lib/activity/validator.go

```diff
@@ -0,0 +1,333 @@
+//   - searchAttributes: Search attributes to validate and unalias
+//   - saMapperProvider: Provider for mapping aliased search attribute names
+//   - saValidator: Validator for search attributes
+//
+// Returns the modified attributes (request ID and unaliased search attributes) if they have been sanitized or an error
+// if validation fails.
+func ValidateStandaloneActivity(
+	activityID string,
+	activityType string,
+	blobSizeLimitError dynamicconfig.IntPropertyFnWithNamespaceFilter,
+	blobSizeLimitWarn dynamicconfig.IntPropertyFnWithNamespaceFilter,
+	inputSizeBytes int,
+	logger log.Logger,
+	maxIDLengthLimit int,
+	namespaceName string,
+	requestID string,
+	searchAttributes *commonpb.SearchAttributes,
+	saMapperProvider searchattribute.MapperProvider,
+	saValidator *searchattribute.Validator) (*ModifiedStandaloneActivityRequestAttributes, error) {
+	modifiedAttributes := &ModifiedStandaloneActivityRequestAttributes{}
+
+	if err := validateRequestID(requestID, maxIDLengthLimit, modifiedAttributes); err != nil {
+		return nil, err
+	}
+
+	if err := validateInputSize(
+		activityID,
+		activityType,
+		blobSizeLimitError,
+		blobSizeLimitWarn,
+		inputSizeBytes,
+		logger,
+		namespaceName); err != nil {
+		return nil, err
+	}
+
+	if searchAttributes != nil {
+		if err := validateAndUnaliasSearchAttributes(
+			modifiedAttributes,
+			namespaceName,
+			searchAttributes,
+			saMapperProvider,
+			saValidator); err != nil {
+			return nil, err
+		}
+	}
+
+	return modifiedAttributes, nil
+}
+
+func validateRequestID(requestID string, maxIDLengthLimit int, modifiedAttributes *ModifiedStandaloneActivityRequestAttributes) error {
+	if requestID == "" {
+		// For easy direct API use, we default the request ID here but expect all SDKs and other auto-retrying clients to set it
+		modifiedAttributes.requestID = uuid.New()
+	}
+
+	if len(requestID) > maxIDLengthLimit || len(modifiedAttributes.requestID) > maxIDLengthLimit {
+		return serviceerror.NewInvalidArgument("RequestID length exceeds limit.")
+	}
+
+	return nil
+}
+
+func validateInputSize(
+	activityID string,
+	activityType string,
+	blobSizeLimitError dynamicconfig.IntPropertyFnWithNamespaceFilter,
+	blobSizeLimitWarn dynamicconfig.IntPropertyFnWithNamespaceFilter,
+	inputSize int,
+	logger log.Logger,
+	namespaceName string) error {
+	sizeWarnLimit := blobSizeLimitWarn(namespaceName)
+	sizeErrorLimit := blobSizeLimitError(namespaceName)
+
+	if inputSize > sizeWarnLimit {
+		logger.Warn("Activity input size exceeds the warning limit.",
+			tag.WorkflowNamespace(namespaceName),
+			tag.ActivityID(activityID),
+			tag.ActivitySize(int64(inputSize)),
+			tag.BlobSizeViolationOperation(activityType))
+
+		if inputSize > sizeErrorLimit {
+			return common.ErrBlobSizeExceedsLimit
+		}
+	}
+
+	return nil
+}
+
+func validateAndUnaliasSearchAttributes(
+	modifiedAttributes *ModifiedStandaloneActivityRequestAttributes,
```

Comment:
Make it a method? I don't personally think the method name needs to encode what it does (i.e. doesn't need to contain "unalias")